test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -0,0 +1,223 @@
+		t.Run(name, func(t *testing.T) {
+			for feature, flag := range test.FeatureGates {
+				defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, feature, flag)()
+			}
+
+			var nodeStrategy testutils.PrepareNodeStrategy
+			nodeStrategy = &testutils.TrivialNodePrepareStrategy{}
+
+			if test.Nodes.NodeAllocatableStrategy != nil {
+				nodeStrategy = test.Nodes.NodeAllocatableStrategy
+			} else if test.Nodes.LabelNodePrepareStrategy != nil {
+				nodeStrategy = test.Nodes.LabelNodePrepareStrategy
+			}
+
+			setupPodStrategy := getPodStrategy(test.InitPods)
+			testPodStrategy := getPodStrategy(test.PodsToSchedule)
+
+			var nodeSpec *v1.Node
+			if test.Nodes.NodeTemplatePath != nil {
+				nodeSpec = &v1.Node{}
+				getSpec(test.Nodes.NodeTemplatePath, nodeSpec)
+			}
+
+			perfScheduling(test.Nodes.Num, test.InitPods.Num, test.PodsToSchedule.Num, nodeStrategy, setupPodStrategy, testPodStrategy, nodeSpec)
+		})
+	}
+}
+
+func perfScheduling(numNodes, numExistingPods, minPods int,
+	nodeStrategy testutils.PrepareNodeStrategy,
+	setupPodStrategy, testPodStrategy testutils.TestPodCreateStrategy,
+	nodeSpec *v1.Node) {
+
+	finalFunc, podInformer, clientset := mustSetupScheduler()
+	defer finalFunc()
+
+	var nodePreparer testutils.TestNodePreparer
+	if nodeSpec != nil {
+		nodePreparer = framework.NewIntegrationTestNodePreparerWithNodeSpec(
+			clientset,
+			[]testutils.CountToStrategy{{Count: numNodes, Strategy: nodeStrategy}},
+			nodeSpec,
+		)
+	} else {
+		nodePreparer = framework.NewIntegrationTestNodePreparer(
+			clientset,
+			[]testutils.CountToStrategy{{Count: numNodes, Strategy: nodeStrategy}},
+			"scheduler-perf-",
+		)
+	}
+
+	if err := nodePreparer.PrepareNodes(); err != nil {
+		klog.Fatalf("%v", err)
+	}
+	defer nodePreparer.CleanupNodes()
+
+	config := testutils.NewTestPodCreatorConfig()
+	config.AddStrategy(setupNamespace, numExistingPods, setupPodStrategy)
+	podCreator := testutils.NewTestPodCreator(clientset, config)
+	podCreator.CreatePods()
+
+	for {
+		scheduled, err := getScheduledPods(podInformer)
+		if err != nil {
+			klog.Fatalf("%v", err)
+		}
+		if len(scheduled) >= numExistingPods {
+			break
+		}
+		klog.Infof("got %d existing pods, required: %d", len(scheduled), numExistingPods)
+		time.Sleep(1 * time.Second)
+	}
+
+	scheduled := int32(0)
+	completedCh := make(chan struct{})
+	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(old, cur interface{}) {
+			curPod := cur.(*v1.Pod)
+			oldPod := old.(*v1.Pod)
+
+			if len(oldPod.Spec.NodeName) == 0 && len(curPod.Spec.NodeName) > 0 {
+				if atomic.AddInt32(&scheduled, 1) >= int32(minPods) {
+					completedCh <- struct{}{}
+				}
+			}
+		},
+	})
+
+	config = testutils.NewTestPodCreatorConfig()
+	config.AddStrategy(testNamespace, minPods, testPodStrategy)
+	podCreator = testutils.NewTestPodCreator(clientset, config)
+	podCreator.CreatePods()
+
+	<-completedCh
+}
```

Comment:
you would have to call b.StopTimer() here once you switch to testing.B

Reply:
Done